src/modules/sensors/voted_sensors_update.cpp

```diff
@@ -921,6 +919,47 @@ bool VotedSensorsUpdate::check_failover(SensorData &sensor, const char *sensor_n
 
 				// reduce priority of failed sensor to the minimum
 				sensor.priority[failover_index] = 1;
+
+				// Update the subsystem_info uORB given that a sensor failed
+				int ctr_valid = 0;
+
+				for (uint8_t i = 0; i < sensor.subscription_count; i++) {
+					if (sensor.priority[i] > 1) { ctr_valid++; }
+
+					PX4_WARN("FAILOVER event (idx=%u)! Sensor %s: Nr. %u Priority: %u", failover_index, sensor_name, i, sensor.priority[i]);
+				}
+
+				PX4_ERR("%s sensor switch from #%i", sensor_name, failover_index);
+
+				if (ctr_valid < 2) { // subsystem_info only contains flags for the first two sensors
+					if (ctr_valid == 0) { // There are no valid sensors left!
+						if (strcmp(sensor_name, "Gyro") == 0) { _info.subsystem_type = subsystem_info_s::SUBSYSTEM_TYPE_GYRO; }
```

Comment:
Can we avoid doing a string comparison here?

Reply:
Yes. I pushed a new commit that implements a "quick and dirty" solution to avoid the strcmp. I think the whole voted_sensors_update module would need some refactoring (like integrating both the sensor_name and sensor_type _inside_ the SensorData class as variables), but i don't feel comfortable to do that. I'd suggest to leave it as is for now and then @bkueng (as the author of that class) could maybe implement that?

Reply:
I agree, as SensorData has grown, a refactoring makes sense